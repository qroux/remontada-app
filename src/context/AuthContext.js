import createDataContext from "./createDataContext";
import strapiApi from "../api/strapiApi";
import AsyncStorage from "@react-native-async-storage/async-storage";

const AuthReducer = (state, action) => {
  switch (action.type) {
    case "SIGNUP":
      return { token: action.payload, errorMsg: "" };
    case "SIGNIN":
      return {
        token: action.payload,
        errorMsg: "",
      };
    case "SIGNOUT":
      return { ...state, token: null };
    case "ADD_ERROR":
      return { ...state, errorMsg: action.payload };
    case "CLEAR_ERROR":
      return { ...state, errorMsg: "" };
    case "LOADED":
      return { ...state, isLoading: false };
    default:
      return state;
  }
};

// actions
const signin = (dispatch) => async ({ email, password }) => {
  try {
    const response = await strapiApi.post("/auth/local", {
      identifier: email,
      password,
    });

    const token = response.data.jwt;
    await AsyncStorage.setItem("remontada_token", token);

    dispatch({
      type: "SIGNIN",
      payload: { email, token },
    });
  } catch (err) {
    dispatch({
      type: "ADD_ERROR",
      payload: err.message,
    });
  }
};

const signup = (dispatch) => async ({ email, password }) => {
  try {
    const response = await strapiApi.post("/auth/local/register", {
      username: email,
      email,
      password,
    });

    const token = response.data.jwt;
    await AsyncStorage.setItem("remontada_token", token);

    dispatch({
      type: "SIGNUP",
      payload: { email, token },
    });
  } catch (err) {
    dispatch({
      type: "ADD_ERROR",
      payload: err.message,
    });
  }
};

const signout = (dispatch) => async () => {
  dispatch({ type: "SIGNOUT" });
  await AsyncStorage.removeItem("remontada_token");
};

//UTILS
const getToken = (dispatch) => async () => {
  const token = await AsyncStorage.getItem("remontada_token");
  if (token) {
    dispatch({ type: "SIGNIN", payload: token });
  }

  dispatch({ type: "LOADED" });
};

const clearError = (dispatch) => async () => {
  dispatch({ type: "CLEAR_ERROR" });
};

export const { Context, Provider } = createDataContext(
  AuthReducer,
  { getToken, clearError, signin, signup, signout },
  { token: null, errorMsg: null }
);
