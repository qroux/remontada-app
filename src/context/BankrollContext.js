import createDataContext from './createDataContext';
import strapiApi from '../api/strapiApi';
import AsyncStorage from '@react-native-async-storage/async-storage';

// REDUCER
const BankrollReducer = (state, action) => {
  switch (action.type) {
    case 'GET_BETS':
      return { ...state, bets: action.payload };
    case 'GET_USER_BANKROLLS':
      return { ...state, bankrolls: action.payload };
    case 'ADD_ERROR':
      return { ...state, errorMsg: action.payload };
    default:
      return state;
  }
};

// ACTIONS
const getBets = (dispatch) => async () => {
  try {
    const response = await strapiApi.get('/bets');
    dispatch({ type: 'GET_BETS', payload: response.data });
  } catch (err) {
    dispatch({ type: 'ADD_ERROR', payload: err });
  }
};

const getUserBankrolls = (dispatch) => async () => {
  try {
    const user_id = await AsyncStorage.getItem('remontada_user_id');
    const response = await strapiApi.get(
      `bankrolls?users_permissions_user=${user_id}`
    );

    dispatch({ type: 'GET_USER_BANKROLLS', payload: response.data });
  } catch (err) {
    dispatch({ type: 'ADD_ERROR', payload: err });
  }
};
// UTILS

// EXPORT
export const { Context, Provider } = createDataContext(
  BankrollReducer,
  { getBets, getUserBankrolls },
  {}
);
