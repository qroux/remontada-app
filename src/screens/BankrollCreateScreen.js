import React from "react";
import { SafeAreaView, Text, View, StyleSheet } from "react-native";
import { Colors, Spacing, Size } from "../assets/main";
import { Common } from "../assets/common";

import { BankrollForm } from "../components/mainComponents/BankrollForm";

export const BankrollCreateScreen = () => (
  <SafeAreaView style={Common.fullPage}>
    <View style={Common.container}>
      <Text style={styles.header}>
        Créer une nouvelle Bankroll pour mesurer vos performances à partir de
        votre mise de départ
      </Text>
      <BankrollForm />
    </View>
  </SafeAreaView>
);

const styles = StyleSheet.create({
  header: {
    fontSize: Size.small,
    fontWeight: "bold",
  },
});
