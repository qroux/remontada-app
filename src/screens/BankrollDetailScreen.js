import React from 'react';
import { Text, View } from 'react-native';

export const BankrollDetailScreen = ({ route }) => (
  <View>
    <Text>BankrollDetailScreen {route.params.bankroll_id}</Text>
  </View>
);
