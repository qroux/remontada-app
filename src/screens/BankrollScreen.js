import { useNavigation } from "@react-navigation/native";
import React, { useEffect, useState } from "react";
import { SafeAreaView, View, Text, StyleSheet } from "react-native";
import { Button, Overlay } from "react-native-elements";
import { Common } from "../assets/common";
import { Colors } from "../assets/main";
import strapiApi from "../api/strapiApi";

import { BankrollList } from "../components/mainComponents/BankrollList";
import { BankrollForm } from "../components/mainComponents/BankrollForm";

export const BankrollScreen = () => {
  const navigation = useNavigation();
  const [visible, setVisible] = useState(false);
  const [bankrolls, setBankrolls] = useState([]);

  useEffect(() => {
    const fetchBankrolls = async () => {
      const getUser = await strapiApi.get("users/me");
      const id = getUser.data._id;

      const response = await strapiApi.get(
        `bankrolls?users_permissions_user=${id}`
      );
      setBankrolls(response.data);
    };

    fetchBankrolls();
  }, []);

  const toggleOverlay = () => {
    setVisible(!visible);
  };

  return (
    <SafeAreaView style={Common.fullPage}>
      <View style={Common.container}>
        <Text style={Common.title}>Bankrolls</Text>
        <BankrollList bankrolls={bankrolls} />
        <Button title="Nouvelle Bankroll" onPress={toggleOverlay} />
      </View>
      <Overlay isVisible={visible} onBackdropPress={toggleOverlay}>
        <BankrollForm toggleOverlay={toggleOverlay} />
      </Overlay>
    </SafeAreaView>
  );
};

const styles = StyleSheet.create({});
