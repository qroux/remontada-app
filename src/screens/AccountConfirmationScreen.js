import React, { useContext } from "react";
import {
  View,
  Text,
  StyleSheet,
  ImageBackground,
  Dimensions,
  SafeAreaView,
  StatusBar,
} from "react-native";
import { Colors, Spacing, Size } from "../assets/main";
import { Context as AuthContext } from "../context/AuthContext";

import { AuthForm } from "../components/authComponents/AuthForm";
import { Footer } from "../components/authComponents/Footer";
import { Button } from "react-native-elements";
import { useNavigation } from "@react-navigation/native";

export const AccountConfirmationScreen = ({ route }) => {
  const navigation = useNavigation();
  const { state, signup } = useContext(AuthContext);
  return (
    <SafeAreaView>
      <ImageBackground
        source={{
          uri:
            "https://images.unsplash.com/photo-1508100134119-f93388e60d95?ixlib=rb-1.2.1&ixid=MXwxMjA3fDB8MHxwaG90by1wYWdlfHx8fGVufDB8fHw%3D&auto=format&fit=crop&w=1234&q=80",
        }}
        style={styles.background}
      >
        <View style={styles.container}>
          <Text style={styles.upTitle}>Compte crée</Text>
          <Text style={styles.header}>Remontad' App</Text>
          <View style={styles.instructionsContainer}>
            <Text style={styles.instructions}>
              Votre compte a été créé. Activez le grâce au lien envoyé par mail
              à l'adresse
            </Text>
            <View style={styles.addressContainer}>
              <Text style={styles.address}>{route.params.email}</Text>
            </View>
          </View>

          <Footer
            header="Vous possédez déjà un compte ?"
            path="Login"
            label="Retour à l'accueil"
          />
        </View>
      </ImageBackground>
    </SafeAreaView>
  );
};

const styles = StyleSheet.create({
  container: {
    padding: Spacing.medium,
    flex: 1,
  },
  background: {
    height: Dimensions.get("window").height - StatusBar.currentHeight,
    width: Dimensions.get("window").width,
    marginTop: StatusBar.currentHeight,
  },
  header: {
    fontSize: Size.large,
    fontWeight: "bold",
    color: "#F2404F",
    marginTop: -15,

    textShadowColor: "black",
  },
  upTitle: {
    color: Colors.textLight,
    fontWeight: "bold",
    fontSize: Size.large,
    marginTop: Spacing.large,
  },
  instructions: {
    fontSize: Size.small,
    color: Colors.textDark,
    fontWeight: "bold",
  },
  instructionsContainer: {
    marginTop: Spacing.large,
    marginBottom: Spacing.small,
    padding: Spacing.medium,
    backgroundColor: Colors.bgLight,
    borderRadius: Size.radius,
  },
  addressContainer: {
    backgroundColor: Colors.bgDark,
    marginVertical: Spacing.medium,
    height: 50,
    alignItems: "center",
    justifyContent: "center",
    borderRadius: Size.radius,
  },
  address: {},
});
