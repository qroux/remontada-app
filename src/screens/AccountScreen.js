import React, { useContext, useEffect, useState } from "react";
import {
  View,
  Text,
  StyleSheet,
  ImageBackground,
  Dimensions,
  SafeAreaView,
  Image,
  FlatList,
} from "react-native";
import strapiApi from "../api/strapiApi";
import { Button } from "react-native-elements";
import { Common } from "../assets/common";
import { Colors, Spacing, Size } from "../assets/main";
import { Context as AuthContext } from "../context/AuthContext";

export const AccountScreen = () => {
  const [teams, setTeams] = useState([]);
  const { state, signout } = useContext(AuthContext);

  useEffect(() => {
    const fetchTeams = async () => {
      const response = await strapiApi.get("/teams");
      setTeams(response.data);
    };

    fetchTeams();
  }, []);
  return (
    <SafeAreaView style={Common.fullPage}>
      <View style={Common.container}>
        <Text style={Common.title}>Account Screen</Text>
        <Button title="disconnect" onPress={signout} />
        <FlatList
          data={teams}
          keyExtractor={(team) => team.name}
          renderItem={({ item }) => {
            return (
              <Image style={styles.logo} source={{ uri: item.logo.url }} />
            );
          }}
        />
      </View>
    </SafeAreaView>
  );
};

const styles = StyleSheet.create({
  logo: {
    height: 100,
    width: 100,
  },
});
