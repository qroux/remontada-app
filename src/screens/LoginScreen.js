import React, { useState, useEffect } from "react";
import AsyncStorage from "@react-native-async-storage/async-storage";
import { useContext } from "react";
import {
  View,
  Text,
  StyleSheet,
  ImageBackground,
  Dimensions,
  SafeAreaView,
  StatusBar,
} from "react-native";
import { Colors, Spacing, Size } from "../assets/main";
import { Context as AuthContext } from "../context/AuthContext";

import { AuthForm } from "../components/AuthForm";
import { Footer } from "../components/Footer";

export const LoginScreen = () => {
  const { state, signin, getToken } = useContext(AuthContext);

  useEffect(() => {
    getToken();
  }, []);

  return (
    <SafeAreaView>
      <ImageBackground
        source={{
          uri:
            "https://images.unsplash.com/photo-1509928015542-fcc9b3bcd048?ixid=MXwxMjA3fDB8MHxwaG90by1wYWdlfHx8fGVufDB8fHw%3D&ixlib=rb-1.2.1&auto=format&fit=crop&w=1234&q=80",
        }}
        style={styles.background}
      >
        <View style={styles.container}>
          <Text style={styles.upTitle}>Bienvenue sur</Text>
          <Text style={styles.header}>Remontad' App</Text>
          <Text style={styles.subTitle}>Connectez vous pour continuer</Text>
          <AuthForm submit={signin} btnTitle="Connexion" />
          <Footer
            header="Pas encore de compte ?"
            path="Register"
            label="Inscription"
            reset={true}
          />
        </View>
      </ImageBackground>
    </SafeAreaView>
  );
};

const styles = StyleSheet.create({
  container: {
    padding: Spacing.medium,
    flex: 1,
  },
  background: {
    height: Dimensions.get("window").height - StatusBar.currentHeight,
    width: Dimensions.get("window").width,
    marginTop: StatusBar.currentHeight,
  },
  header: {
    fontSize: Size.large,
    fontWeight: "bold",
    color: "#F2404F",
    marginTop: -15,
    textShadowColor: "black",
  },
  subTitle: {
    fontSize: Size.regular,
    color: "white",
    fontWeight: "bold",
    marginTop: Spacing.large,
    marginBottom: Spacing.small,
  },
  upTitle: {
    color: Colors.textLight,
    fontWeight: "bold",
    fontSize: Size.large,
    marginTop: Spacing.large,
  },
});
