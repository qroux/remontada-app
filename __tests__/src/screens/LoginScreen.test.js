import React, { useContext, useNavigation } from 'react';
import renderer from 'react-test-renderer';
import ShallowRenderer from 'react-test-renderer/shallow';
import { LoginScreen } from '../../../src/screens/LoginScreen';

import { Provider as AuthProvider } from '../../../src/context/AuthContext';
import { Context as AuthContext } from '../../../src/context/AuthContext';

// let realUseContext;
// let useContextMock;
// // // Setup mock
// beforeEach(() => {
//   realUseContext = React.useContext;
//   // useContextMock = React.useContext = jest.fn();
//   useContextMock = useContext(AuthContext);
// });
// // Cleanup mock
// afterEach(() => {
//   React.useContext = realUseContext;
// });

// test("mock hook", () => {
//   useContextMock.mockReturnValue("Test Value");
//   const element = new ShallowRenderer().render(<MyComponent />);
//   expect(element.props.children).toBe("Test Value");
// });

// describe("<LoginScreen />", () => {
//   it("has 1 child", async () => {
//     const tree = renderer.create(<LoginScreen />).toJSON();
//     expect(tree.children.length).toBe(1);
//   });
// });

it('renders correctly', () => {
  const tree = renderer
    .create(
      <AuthProvider>
        <LoginScreen />
      </AuthProvider>
    )
    .toJSON();
  expect(tree).toMatchSnapshot();
});
