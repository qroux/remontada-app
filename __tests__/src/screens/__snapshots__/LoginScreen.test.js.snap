// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders correctly 1`] = `
<RCTSafeAreaView
  emulateUnlessSupported={true}
>
  <View
    accessibilityIgnoresInvertColors={true}
    style={
      Object {
        "height": 1334,
        "marginTop": null,
        "width": 750,
      }
    }
  >
    <Image
      source={
        Object {
          "uri": "https://images.unsplash.com/photo-1509928015542-fcc9b3bcd048?ixid=MXwxMjA3fDB8MHxwaG90by1wYWdlfHx8fGVufDB8fHw%3D&ixlib=rb-1.2.1&auto=format&fit=crop&w=1234&q=80",
        }
      }
      style={
        Array [
          Object {
            "bottom": 0,
            "left": 0,
            "position": "absolute",
            "right": 0,
            "top": 0,
          },
          Object {
            "height": 1334,
            "width": 750,
          },
          undefined,
        ]
      }
    />
    <View
      style={
        Object {
          "flex": 1,
          "padding": 20,
        }
      }
    >
      <Text
        style={
          Object {
            "color": "white",
            "fontSize": 45,
            "fontWeight": "bold",
            "marginTop": 50,
          }
        }
      >
        Bienvenue sur
      </Text>
      <Text
        style={
          Object {
            "color": "#F2404F",
            "fontSize": 45,
            "fontWeight": "bold",
            "marginTop": -15,
            "textShadowColor": "black",
          }
        }
      >
        Remontad' App
      </Text>
      <Text
        style={
          Object {
            "color": "white",
            "fontSize": 24,
            "fontWeight": "bold",
            "marginBottom": 5,
            "marginTop": 50,
          }
        }
      >
        Connectez vous pour continuer
      </Text>
      <View>
        <View
          style={
            Object {
              "height": 50,
              "marginBottom": 10,
              "paddingBottom": 0,
              "paddingHorizontal": 0,
              "width": "100%",
            }
          }
        >
          <View
            style={
              Object {
                "alignItems": "center",
                "backgroundColor": "white",
                "borderBottomWidth": 1,
                "borderColor": "#f4f4f5",
                "borderRadius": 3,
                "borderWidth": 1,
                "flexDirection": "row",
                "height": 50,
                "paddingHorizontal": 10,
                "transform": Array [
                  Object {
                    "translateX": 0,
                  },
                ],
              }
            }
          >
            <TextInput
              allowFontScaling={true}
              autoCapitalize="none"
              autoCorrect={false}
              editable={true}
              onChangeText={[Function]}
              placeholder="email"
              placeholderTextColor="#86939e"
              rejectResponderTermination={true}
              style={
                Object {
                  "alignSelf": "center",
                  "color": "#242424",
                  "flex": 1,
                  "fontSize": 18,
                  "minHeight": 40,
                }
              }
              testID="RNE__Input__text-input"
              underlineColorAndroid="transparent"
              value=""
            />
          </View>
          <Text
            style={
              Object {
                "color": "#ff190c",
                "fontSize": 12,
                "margin": 5,
              }
            }
          />
        </View>
        <View
          style={
            Object {
              "height": 50,
              "marginBottom": 10,
              "paddingBottom": 0,
              "paddingHorizontal": 0,
              "width": "100%",
            }
          }
        >
          <View
            style={
              Object {
                "alignItems": "center",
                "backgroundColor": "white",
                "borderBottomWidth": 1,
                "borderColor": "#f4f4f5",
                "borderRadius": 3,
                "borderWidth": 1,
                "flexDirection": "row",
                "height": 50,
                "paddingHorizontal": 10,
                "transform": Array [
                  Object {
                    "translateX": 0,
                  },
                ],
              }
            }
          >
            <TextInput
              allowFontScaling={true}
              autoCapitalize="none"
              autoCorrect={false}
              editable={true}
              onChangeText={[Function]}
              placeholder="mot de passe"
              placeholderTextColor="#86939e"
              rejectResponderTermination={true}
              secureTextEntry={true}
              style={
                Object {
                  "alignSelf": "center",
                  "color": "#242424",
                  "flex": 1,
                  "fontSize": 18,
                  "minHeight": 40,
                }
              }
              testID="RNE__Input__text-input"
              underlineColorAndroid="transparent"
              value=""
            />
            <View
              style={
                Object {
                  "alignItems": "center",
                  "height": 40,
                  "justifyContent": "center",
                  "marginVertical": 4,
                  "paddingRight": 4,
                }
              }
            >
              <Text />
            </View>
          </View>
          <Text
            style={
              Object {
                "color": "#ff190c",
                "fontSize": 12,
                "margin": 5,
              }
            }
          />
        </View>
        <View
          style={
            Object {
              "borderRadius": 3,
              "overflow": "hidden",
            }
          }
        >
          <View
            accessibilityRole="button"
            accessibilityState={
              Object {
                "busy": false,
                "disabled": false,
              }
            }
            accessible={true}
            focusable={true}
            onClick={[Function]}
            onResponderGrant={[Function]}
            onResponderMove={[Function]}
            onResponderRelease={[Function]}
            onResponderTerminate={[Function]}
            onResponderTerminationRequest={[Function]}
            onStartShouldSetResponder={[Function]}
            style={
              Object {
                "opacity": 1,
              }
            }
          >
            <View
              style={
                Object {
                  "alignItems": "center",
                  "backgroundColor": "#2089dc",
                  "borderColor": "#2089dc",
                  "borderRadius": 3,
                  "borderWidth": 0,
                  "flexDirection": "row",
                  "height": 50,
                  "justifyContent": "center",
                  "marginTop": 5,
                  "padding": 8,
                }
              }
            >
              <Text
                style={
                  Object {
                    "color": "white",
                    "fontSize": 18,
                    "paddingVertical": 1,
                    "textAlign": "center",
                  }
                }
              >
                Connexion
              </Text>
            </View>
          </View>
        </View>
      </View>
      <View
        style={
          Object {
            "alignItems": "center",
            "flex": 1,
            "justifyContent": "center",
            "marginTop": 50,
          }
        }
      >
        <View
          accessible={true}
          focusable={true}
          onClick={[Function]}
          onResponderGrant={[Function]}
          onResponderMove={[Function]}
          onResponderRelease={[Function]}
          onResponderTerminate={[Function]}
          onResponderTerminationRequest={[Function]}
          onStartShouldSetResponder={[Function]}
          style={
            Object {
              "opacity": 1,
            }
          }
        >
          <Text
            style={
              Object {
                "color": "white",
                "fontSize": 24,
                "fontWeight": "bold",
              }
            }
          >
            Inscription
          </Text>
        </View>
        <View
          accessible={true}
          focusable={true}
          onClick={[Function]}
          onResponderGrant={[Function]}
          onResponderMove={[Function]}
          onResponderRelease={[Function]}
          onResponderTerminate={[Function]}
          onResponderTerminationRequest={[Function]}
          onStartShouldSetResponder={[Function]}
          style={
            Object {
              "opacity": 1,
            }
          }
        >
          <Text
            style={
              Object {
                "color": "white",
                "fontSize": 24,
                "fontWeight": "bold",
              }
            }
          >
            Mot de passe oubli√©
          </Text>
        </View>
      </View>
    </View>
  </View>
</RCTSafeAreaView>
`;
